kernel/Base/Node.h

```diff
@@ -155,11 +159,19 @@ class Node {
         setLocalNumberOfBasisFunctions(number);
     }
 
-    bool isOwnedByCurrentProcessor() const;
+    bool isOwnedByCurrentProcessor() const final;
 
     /// The element owning this node, only valid if the node is owned by the
     /// current processor
-    Element *getOwningElement() const;
+    const Element *getOwningElement() const final;
+
+    void visitEntity(MeshEntityVisitor<> &visitor) final {
+        visitor.visit(*this);
+    }
+
+    void visitEntity(MeshEntityVisitor<true> &visitor) const final {
+        visitor.visit(*this);
+    }
```

Comment:
I am not sure the performance advantage is worth the readability, probably has to be benchmarked.

Reply:
What would you then suggest for the readability? Because I think this boiler plate is needed for a visitor pattern.

Reply:
Okay I would write visitor.visit(node) in the code but this is also fine

Reply:
But what would node then refer to?